Source/WebCore/rendering/RenderLayerScrollableArea.cpp

```diff
@@ -2052,5 +2054,23 @@ void RenderLayerScrollableArea::invalidateScrollAnchoringElement()
         m_scrollAnchoringController->invalidateAnchorElement();
 }
 
+void RenderLayerScrollableArea::registerScrollbars()
+{
+    setScrollbarsController(nullptr);
+    createScrollbarsController();
+    if (auto* horizontalScrollbar = this->horizontalScrollbar()) {
+        horizontalScrollbar->initScrollerImps();
+        didAddScrollbar(horizontalScrollbar, ScrollbarOrientation::Horizontal);
+    }
+    if (auto* verticalScrollbar = this->verticalScrollbar()) {
+        verticalScrollbar->initScrollerImps();
+        didAddScrollbar(verticalScrollbar, ScrollbarOrientation::Vertical);
+    }
+}
+
+bool RenderLayerScrollableArea::shouldRenderScollbarsInProcess()
+{
+    return m_layer.isInsideSVGForeignObject();
```

Comment:
Isn't it really about not being able to use asyncScrolling, i.e it will have no ScrollingNodeID? So is this the same as when `RenderLayerScrollableArea::usesAsyncScrolling()` returns false? Or perhaps `RenderLayerScrollableArea::canUseCompositedScrolling()` ?

Reply:
Right, using `RenderLayerScrollableArea::usesAsyncScrolling` simplifies the patch quite a bit since we don't have to set it from `RenderLayer::updateAncestorDependentState`, we can just query `RenderLayerScrollableArea::usesAsyncScrolling` at scrollbar controller creation time.